rocAL/rocAL_pybind/examples/rocAL_api_coco_pipeline.py

```diff
@@ -107,251 +65,165 @@ def next(self):
-
-        image_id_tensor = torch.tensor(self.image_id)
-        image_size_tensor = torch.tensor(self.img_size).view(-1, self.bs, 2)
-        for i in range(self.bs):
-            index_list = []
-            actual_bboxes = []
-            actual_labels = []
-            for idx, x in enumerate(encodded_labels_tensor[i]):
-                if x != 0:
-                    index_list.append(idx)
-                    actual_bboxes.append(encoded_bboxes_tensor[i][idx].tolist())
-                    actual_labels.append(encodded_labels_tensor[i][idx].tolist())
-
-            if self.display:
-                img = self.out
-                draw_patches(img[i], self.image_id[i],
-                             actual_bboxes, self.device)
+            self.output_tensor_list = self.loader.getOutputTensors()
+
+        if self.out is None:
+            self.w, self.h = self.output_tensor_list[0].max_shape()
+            self.bs = self.output_tensor_list[0].batch_size()
+            self.color_format = self.output_tensor_list[0].color_format() if self.color_format else self.color_format
+            torch_gpu_device = torch.device('cuda', self.device_id)
+            if self.tensor_format == types.NCHW:
+                torch_gpu_device = torch.device('cuda', self.device_id)
+                if self.tensor_dtype == types.FLOAT:
+                    self.out = torch.empty((self.batch_size, self.color_format, self.h, self.w,), dtype = torch.float32, device = torch_gpu_device)
+                elif self.tensor_dtype == types.FLOAT16:
+                    self.out = torch.empty((self.batch_size, self.color_format, self.h, self.w,), dtype = torch.float16, device = torch_gpu_device)
+                elif self.tensor_dtype == types.UINT8:
+                    self.out = torch.empty((self.batch_size, self.color_format, self.h, self.w,), dtype = torch.uint8, device = torch_gpu_device)
+            else:  # NHWC
+                torch_gpu_device = torch.device('cuda', self.device_id)
+                if self.tensor_dtype == types.FLOAT:
+                    self.out = torch.empty((self.batch_size, self.h, self.w, self.color_format), dtype = torch.float32, device = torch_gpu_device)
+                elif self.tensor_dtype == types.FLOAT16:
+                    self.out = torch.empty((self.batch_size, self.h, self.w, self.color_format), dtype = torch.float16, device = torch_gpu_device)
+                elif self.tensor_dtype == types.UINT8:
+                    self.out = torch.empty((self.batch_size, self.h, self.w, self.color_format), dtype = torch.uint8, device = torch_gpu_device)
+            self.labels_tensor = torch.empty(self.batch_size, dtype = torch.int32, device = torch_gpu_device)
+
+        self.output_tensor_list[0].copy_data(ctypes.c_void_p(self.out.data_ptr()))
+
+        labels_array = self.loader.GetBoundingBoxLabels()
+        encodded_labels_tensor = []
+        encoded_bboxes_tensor = []
+        for label in labels_array:
```

Comment:
we can't use for loop here as it will copy each label seperate. We need to get rid of as many data conversion code as possible in python so our performance is not affected. Please return the labels in the expected format and use labels_array directly here

Reply:
This looping is required for RetinaNet Rajy & this happens only for batchSize number of times.
This is done to match the expected output format required in training